projects/cdk/schematics/ng-update/v4/migrate-addon-doc/index.ts

```diff
@@ -0,0 +1,54 @@
+import type {Rule, SchematicContext, Tree} from '@angular-devkit/schematics';
+import {chain} from '@angular-devkit/schematics';
+import {replaceIdentifiers} from '@taiga-ui/cdk/schematics/ng-update/steps';
+import {getSourceFiles, saveActiveProject} from 'ng-morph';
+
+import {ALL_TS_FILES} from '../../../constants/file-globs';
+import type {TuiSchema} from '../../../ng-add/schema';
+import {
+    FINISH_SYMBOL,
+    infoLog,
+    REPLACE_SYMBOL,
+    SMALL_TAB_SYMBOL,
+    titleLog,
+} from '../../../utils/colored-log';
+import {removeModules} from '../../steps/remove-module';
+import {getFileSystem} from '../../utils/get-file-system';
+import {DOC_SYMBOLS_TO_REPLACE} from './replace-symbols';
+
+export function migrateAddonDoc(options: TuiSchema): Rule {
+    return chain([
+        (tree: Tree, _context: SchematicContext) => {
```

Comment:
let's add early return if there is no addon-doc package in package.json

Reply:
fixed